core/modules/layout/layout.admin.inc

```diff
@@ -281,23 +281,89 @@ function layout_settings_form($form, &$form_state, Layout $layout) {
     }
   }
 
-  $items = array(
+  // Path entry field with dynamic information about the entered path and
+  // toggleable placeholder examples.
+  $path = isset($form_state['values']['path']) ? $form_state['values']['path'] : $layout->getPath();
+  $is_new = isset($layout->is_new) && $layout->is_new;
+  $has_placeholder = strpos($path, '%') !== FALSE;
+  if (empty($path)) {
+    $path_matches_text = t('Please enter a path.');
+  }
+  else {
+    $path_matches = db_query('
+      SELECT path, page_callback
+      FROM {menu_router}
+      WHERE path = :path
+      ', array(':path' => $path))
+      ->fetchAllKeyed();
```

Comment:
Could you used `menu_get_item()` here instead of a direct database query? Seems like we should have some function for loading menu items by router path. If not, perhaps put this in a private function?

Reply:
`menu_get_item()` is not the right tool; it takes a normal path (`node/%`) and returns the menu router path (`node/%`). This function is checking for the existence of the menu router path itself. So a function it will be.

I created `_layout_menu_router_path_exists()` for this function. But it looks like something that might be useful elsewhere, so I raise the question of whether this should be a public function in `menu.inc`, perhaps called `menu_router_path_exists()`?

Reply:
I think leaving it private unless we need it in multiple places seems fine.

Reply:
OK!